### PR TITLE
chore(resources): rename service-info to metadata

### PR DESCRIPTION
**Pull request by orhun**
## Description of change

Rename `shuttle-service-info` to `shuttle-metadata` since it will contain more information than only the service in the future.

## How has this been tested? (if applicable)

Na.

### CODE REVIEW

**Comment by orhun** [comment]
I didn't rename the internal types as requested @jonaro00

**Review by jonaro00 — APPROVED** [review]

File: resources/metadata/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Shuttle Service Info
```
Comment (oddgrd) [inline resources/metadata/README.md:1]: Should this header also be renamed to Shuttle Metadata?

Comment (orhun) [reply resources/metadata/README.md]: Done!

File: resources/metadata/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+## Usage
+
+Add `shuttle-metadata` to the dependencies for your service.
+
+You can get this resource using the `shuttle_metadata::ShuttleServiceInfo` attribute to get a `ServiceInfo`. This struct will contain information such as the Shuttle service name.
+
+```rust
+#[shuttle_runtime::main]
+async fn app(
+    #[shuttle_metadata::ShuttleServiceInfo] service_info: shuttle_metadata::ServiceInfo,
```
Comment (jonaro00) [inline resources/metadata/README.md:14]: We can also rename the structs

**Comment by iulianbarbu** [comment]
We have a regression in `main` regarding a clippy lint, maybe you can address that with the PR too, to have a green CI: https://app.circleci.com/pipelines/github/shuttle-hq/shuttle/3891/workflows/d88406e0-d280-44ba-9f95-d1f8f7420677/jobs/80954

File: resources/metadata/README.md (hunk 2 above)
Comment (orhun) [reply resources/metadata/README.md]: done in 1c2aff749a266e2f5cbfa91f296b933822b6e97b

**Comment by orhun** [comment]
@iulianbarbu addressed in 566b50ab79cb5ef5a16d60f47778f67a3ea5f6a6

**Review by chesedo — APPROVED** [review]
Nice, thanks @orhun!

**Pull request merged**